packages/vite/src/node/build.ts

```diff
@@ -343,6 +343,10 @@ async function doBuild(
         ) as DepOptimizationMetadata
         knownImports = Object.keys(data.optimized)
       } catch (e) {}
+      if (knownImports) {
+        knownImports = knownImports.map(getNpmPackageName)
+        knownImports = unique(knownImports)
```

Comment:
**thought (non-blocking):** Are these two lines the same as 
```ts
knownImports = unique(knownImports.map(getNpmPackageName))
```

Maybe this is better, maybe not :shrug: I'm not sure myself

Reply:
Yes both are equivalent. I have a slight preference for the two-lines version. But I can see it to be an unsual preference so I guess we should go for your one-liner proposal?

Reply:
Let's bring a third perspective to help :slightly_smiling_face: 

@patak-js What would you prefer?

Reply:
On second thought, it seems to me that the one-liner fits better Vite's codebase style. But I don't know Vite's source much so not sure here. I went ahead and commited the one-liner. Curious to see what patak-js is going to say. Thank you for your review btw.

Reply:
I also prefer the one-liner here, current debug tools have no issue with placing breakpoints in the middle of that expression. About the style in the rest of the code base, I saw examples of both.

Reply:
👍 Done.